ext/mysqli/mysqli_warning.c

```diff
@@ -129,7 +129,7 @@ MYSQLI_WARNING * php_get_warnings(MYSQLND_CONN_DATA * mysql)
 
 	for (;;) {
 		zval *entry;
-		int errno;
+		int php_mysqli_errno;
```

Comment:
Curious, why are these renames necessary?

Reply:
Or rather, I can see why it is necessary, the question is more why `-Wstrict-prototypes` specifically makes it necessary. Does it contain some special check for errno?

Regarding the change itself, I don't think we should have a php_ prefix for local variables. This should be just mysqli_errno, or else some variation on the name (err_num).

Reply:
Can someone explain it to me also? Why are we renaming these?

Reply:
`errno` is a global variable/macro defined by the implementation, as such redefining its prototype (here saying it is an `int`) generates a Wstrict-prototypes warning.
Moreover, overridding the value of errno might have unintended consequence as you are working with a global value (although very unlikely here).

I'll rename the usage here and in LDAP after I merged rebased on the void changes.

Reply:
Isn't this a bug then? Should it not be fixed separately on its own and in 7.4?